libraries/core_libs/network/include/network/tarcap/taraxa_peer.hpp

```diff
@@ -86,6 +86,8 @@ class TaraxaPeer : public boost::noncopyable {
   std::atomic<bool> syncing_ = false;
   std::atomic<uint64_t> dag_level_ = 0;
   std::atomic<uint64_t> pbft_chain_size_ = 0;
+  // CONCERN: period different from chain size? how?
```

Comment:
actually period should always be: `chain_size + 1 == period`

Reply:
@JakubFornadel  is that true if we have seen 2t+1 cert votes for a block but don't yet have the block to add to the chain?   Don't we now advance to the next period without getting the block?    ... And I think actually we still need to add in check that (period == chain_size + 1) within pbft mgr before participating within the current period by voting.

Reply:
@JakubFornadel  I added 

`assert(pbft_chain_->getPbftChainSize() + 1 == period_);`

into stateOperations_() in pbft and this assertion does not always pass.

Reply:
@justinsnapp 

- no we never advance to the next period until we actually push the block into the chain...
- that assert should not be there: pbft manager runs independently in its own thread and increasing chain / increasing pbft period is not atomic operation. We increment chain size, which does some db writes etc... and then we increment pbft period so there is always a chance that stateOperations_() is called between these 2 operations and such assert would fail

Reply:
Actually, I just realized we dont need `period_` at all as it is always incremented only if chain_size is increment, we can simply use chain_size + 1 as current_pbft_period and thats it -> in such case even the race condition I mentioned above will disappear